Adding cashier closing feature

elba7r opened the pull request:
As it is very important for the cashiers to have a cashier closing feature i have added the doctype at the accounts module and put a link for it at the pos menu also a link for it at the accounts py config file to show it at the accounts module page .

The cashier closing takes the date time and the user automatically, the user can add the expenses which is taken from his save and add the custody which he received at the beginning of his shift also he add the amount which is at his save at the end of the shift and the system will calculate the net total amount which is supposed to be equal to the total of his sales invoices.

elba7r committed:
adding cashier closing feature

cashier closing feature has been added as now the cashier can make a
closure to record every thing from his pos as the expenses which has
been paid and the custody which he had at the starting and the amount
which is in his save and the system will record the date and time and
the user and the total net amount automatically which supposed be equal
to all of his sales invoices totals

elba7r committed:
add cashier closing link at pos menu

elba7r committed:
add cashier closing link to accounts config file

adding cashier closing link at accounts config file to show it at the
accounts module page

rmehta commented on erpnext/accounts/doctype/cashier_closing/cashier_closing.js:
```diff
@@ -0,0 +1,18 @@
+
+frappe.ui.form.on('Cashier Closing', {
+
+	expense: function(frm){
```
all calculations must be on server side

https://github.com/frappe/erpnext/wiki/Coding-Standards#all-business-logic-must-be-api-friendly-and-must-be-implemented-in-server-side-python

elba7r replied:
now the calculations are on the server side

rmehta reviewed (changes requested):
Please also add screenshots for UX

https://github.com/frappe/erpnext/wiki/Pull-Request-Checklist#user-experience--animated-gif

nabinhait commented:
@elba7r How this is different from "POS Closing Voucher" feature in develop branch?
https://github.com/frappe/erpnext/pull/14270

elba7r committed:
adding more features to cashier closing

adding from time to the cashier closing in case the cashier working more
than one shift also giving him the ability to add the type of payments
to his cashier closure ,
also adding validation to be sure the from time is less than to time.

elba7r commented:
@rmehta  @nabinhait 

- Now all calculations are at the server side
- The Current "POS Closing Voucher"  is working as a summary for the cashier invoices and i believe it is not the aim from the cashier closing as i can get that information from the reporting of the sales invoices, but the cashier closing needed for the cashier him self to be able to write the values by him self then the manager come and review and compare it with the invoices recorded in the data base to be able to find the difference  if found .
-cashier closing feature get the date automatic and the user also, the cashier set the From Date and To Date by him self ( in case he worked more than one shift )
-he also write the expenses which token from his save and the custody which he got at the beginning of his shift ( small changes to help him giving change to the customer or any else) 
-he also add the amount of money based on mode of payments ( cash - credit card ... etc)
-the system calculate the outstanding amounts on the period ( from time - to time ) for the user and add it automatically 
The system making the calculations ( net_amount = payments + expenses + outstanding - custody ) and it should equal the total of his sales invoices, 
so if the manager found any differences he can track and check for the reasons

![cashier closing 1](https://user-images.githubusercontent.com/23710710/44980433-2cf12800-af70-11e8-9436-b4fdab01a2ec.png)
![cashier closing 2](https://user-images.githubusercontent.com/23710710/44980434-2cf12800-af70-11e8-9b5d-1d0a997b37b0.png)

nabinhait committed:
Update cashier_closing.py

yashodhank commented:
@zkhan44499 This is not right place to seek help, go to https://discuss.erpnext.com and ask for help or hire service provider https://erpnext.org/service-providers.

f-9t9it commented:
Hello,

We have made Enhanced Pos Closing Voucher, Using a method for Cash Opening + Sales - Returns - Cash Change given = Closing Cash and also given a Manual Fetch button to fetch updated sales invoice incase the user opened pos closing voucher in the middle of the day.

I can share our Pos Closing Voucher screenshots and also we designed receipt printer size print format. 
Maybe this can help.
In our next pos closing voucher update will have expense added to it.

nabinhait commented:
@f-9t9it Cool 👍 Let's send the PR with some screenshots.

f-9t9it commented:
![whatsapp image 2019-01-31 at 1 05 52 pm](https://user-images.githubusercontent.com/39218579/52048448-7912b980-255c-11e9-87ec-e6e3dcffcffd.jpeg)
![pos closing -1](https://user-images.githubusercontent.com/39218579/52048722-238adc80-255d-11e9-8da7-d911ff4d16d0.png)
![pos closing -2](https://user-images.githubusercontent.com/39218579/52048888-7c5a7500-255d-11e9-8bc7-01ab865f2b51.png)

Kindly find some for now. Need to edit 2 more.

f-9t9it commented:
![pos closing - fetch updates](https://user-images.githubusercontent.com/39218579/52049008-c3e10100-255d-11e9-97f8-c5418caa073e.png)

![pos closing - print format](https://user-images.githubusercontent.com/39218579/52049066-e410c000-255d-11e9-8d11-008ed9ab9632.png)
